docs/docs/docs/01-core/configuration/00-intro.md

```diff
@@ -0,0 +1,115 @@
+
+## Base Strapi Configurations
+
+The following are considered 'base' Strapi configurations and a description of what each file contains is given, so that new settings can be placed in the appropriate place.
+
+A Strapi configuration file must export either an object, or a function that returns an object.
+
+WARNING: There are some discrepancies where certain configuration files such as middlewares may return an array and some files may not be able to export a function. Please confirm before use, and update this documentation when exceptions are found.
+
+(NOT YET IMPLEMENTED) Starting in Strapi v5, the contents of the base configurations will be strictly defined and no values that are not created by Strapi will be allowed in them; that is, an error will be thrown on unrecognized values. Any user additions must go in separate configuration namespaces.
+
+### Base Config Names
+
+#### admin
+
+Defines settings related to the Strapi admin panel, such as the 'autoopen' option for `yarn develop`
+
+#### server
+
+Defines settings related to the Strapi backend server, such as 'host' and 'port'
+
+#### database
+
+Defines the database configuration options.
+
+NOTE: Most of this is defined by Knex, be careful when adding options that they are compatible or placed in the appropriate location
+
+#### api
+
+Content API configuraiton, for example the 'rest.maxLimit' option to set the maximum `limit` available in a query.
+
+#### features
+
+Feature flags for enabling and configuring future features that would be breaking changes and cannot be enabled by default, or experimental features that are not ready for public stable release.
+
+#### plugins
+
+Contains plugin configurations, with each root-level value the id of a plugin, for example, 'users-permissions'
+
+IMPORTANT: Plugins is an exception to the normal namespacing in config.get(). An example plugins.js file has this shape:
+
+```
+module.exports = () => ({
+  graphql: {
+    enabled: true,
+    config: {
+      endpoint: '/graphql',
+
+      defaultLimit: 25,
+      maxLimit: 100,
+
+      apolloServer: {
+        tracing: true,
+      },
+    },
+  },
+});
+```
+
+However, the `config` attribute is flattened and removed so that its values are accessible at, for example, `strapi.config.get('plugins.graphql.endpoint')`
+
+NOTE: Because this configuration contains user and plugin keys and Strapi does not yet provide a method to extend the definitions, this configuration is not strictly typed and will not be loaded from the environment variables
+
+#### middlewares
+
+The configuration of middlewares.
+
+NOTE: This is the only configuration that is currently an array rather than an object. It also very loosely typed. Future improvements are planned to strongly type all the built-in Strapi middlewares.
```

Comment:
```suggestion
NOTE: This is the only configuration that is currently an array rather than an object. It is also very loosely typed. Future improvements are planned to strongly type all the built-in Strapi middlewares.
```